larq_zoo/literature/meliusnet.py

```diff
@@ -0,0 +1,240 @@
+        f_in = x.shape.as_list()[-1]
+        return tf.math.add(x, tf.pad(w, [[0, 0], [0, 0], [0, 0], [f_in - 64, 0]]))
+
+    def transition_block(self, x, filters):
+        x = self.norm(x)
+        x = self.pool(x)
+        x = self.act(x)
+        x = tf.keras.layers.Conv2D(
+            filters,
+            1,
+            use_bias=self.use_conv_bias,
+            kernel_initializer=self.kernel_initializer,
+        )(x)
+        return x
+
+    def block(self, x):
+        x = self.dense_block(x)
+        x = self.improvement_block(x)
+        return x
+
+    def build(self) -> tf.keras.models.Model:
+        x = self.input_tensor
+        x = self.group_stem(x)
+        for i, (n, f) in enumerate(zip(self.num_blocks, self.transition_features)):
+            for j in range(n):
+                x = self.block(x)
+            if f:
+                x = self.transition_block(x, f)
+
+        x = self.norm(x)
+        x = self.act(x)
+        x = tf.keras.layers.GlobalAvgPool2D()(x)
+        x = tf.keras.layers.Dense(
+            self.num_classes,
+            activation="softmax",
+            kernel_initializer=self.kernel_initializer,
+        )(x)
+
+        model = tf.keras.models.Model(
+            inputs=self.input_tensor, outputs=x, name=self.name
+        )
+
+        if self.weights:
+            weights = self.weights
+
+            if weights.startswith("gs://"):
+                model.load_weights(utils.get_gcp_weights(weights))
+            else:
+                model.load_weights(weights)
+
+        return model
+
+
+@factory
+class MeliusNet22Factory(MeliusNetFactory):
+    num_blocks = (4, 5, 4, 4)
+    transition_features = (160, 224, 256, None)
+    name = "meliusnet22"
+
+    def build(self):
+        model = super().build()
+
+        # Load weights.
+        if self.weights == "imagenet":
+            # Download appropriate file
+            if self.include_top:
+                weights_path = utils.download_pretrained_model(
+                    model="meliusnet22",
+                    version="v0.1.0",
+                    file="meliusnet22_weights.h5",
+                    file_hash="",  # TODO
+                )
+            else:
+                weights_path = utils.download_pretrained_model(
+                    model="meliusnet22",
+                    version="v0.1.0",
+                    file="meliusnet22_weights_notop.h5",
+                    file_hash="",  # TODO
+                )
+            model.load_weights(weights_path)
+        elif self.weights is not None:
+            model.load_weights(self.weights)
+        return model
+
+
+def MeliusNet22(
+    *,  # Keyword arguments only
+    input_shape: Optional[Sequence[Optional[int]]] = None,
+    input_tensor: Optional[tf.Tensor] = None,
+    weights: Optional[str] = "imagenet",
+    include_top: bool = True,
+    num_classes: int = 1000,
+) -> tf.keras.models.Model:
+    """Instantiates the MeliusNet22 architecture.
```

Comment:
I know this is just a draft, but would you mind adding the final validation accuracy to the docstring? (see #125)